common/app/common/prices/PriceFinder.java

```diff
@@ -0,0 +1,112 @@
+package common.prices;
+
+import com.neovisionaries.i18n.CountryCode;
+import io.sphere.sdk.channels.Channel;
+import io.sphere.sdk.customergroups.CustomerGroup;
+import io.sphere.sdk.models.Reference;
+import io.sphere.sdk.products.Price;
+
+import javax.money.CurrencyUnit;
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.asList;
+
+public class PriceFinder {
+
+    private PriceFinder() {
+
+    }
+
+    public static PriceFinder of() {
+        return new PriceFinder();
+    }
+
+    /**
+     * currency should always match
+     country, customer group and channel.
+     customer group and channel
+     customer group and country
+     customer group
+     country and channel
+     channel
+     country
+     any left
+     */
+    public Optional<Price> findPrice(final Collection<Price> prices,
```

Comment:
I would suggest to initialize a pricefinder with final CurrencyUnit currency,
+                                     final CountryCode country,
+                                     final java.util.Optional<Reference<CustomerGroup>> customerGroup,
+                                     final java.util.Optional<Reference<Channel>> channel,
+                                     final ZonedDateTime userTime

So you can easily reuse it for multiple products, for example on the pop.

So the price finding has the signature: 
`public Optional<Price> findPrice(final List<Price> prices);`
I am not sure if we can use Collection, since a Set would may produce weird results.